src/SKVClient/src/skv/dto/SKVRecord.cpp

```diff
@@ -266,11 +268,23 @@ SKVRecord SKVRecord::getSKVKeyRecord() {
         throw KeyNotAvailableError("Cannot create an SKVKeyRecord from record without key values");
     }
 
+    size_t num_keys = schema->partitionKeyFields.size() + schema->rangeKeyFields.size();
     SKVRecordBuilder builder(collectionName, schema);
     MPackReader reader(storage.fieldData);
+    std::vector<bool> excludedFields = storage.excludedFields;
+
+    // If size() == 0 then all fields were included, so resize the key_storage excluded fields
+    if (excludedFields.size() == 0) {
+        excludedFields.resize(schema->fields.size(), false);
+    }
+
+    // Exclude all value fields
+    for (size_t i = num_keys; i < excludedFields.size(); ++i) {
+        excludedFields[i] = true;
+    }
 
-    for (size_t i = 0; i < schema->partitionKeyFields.size() + schema->rangeKeyFields.size(); ++i) {
-        if (!storage.excludedFields[i]) {
+    for (size_t i = 0; i < excludedFields.size(); ++i) {
```

Comment:
this should only iterate up to num_keys, not the full number of fields

Reply:
Non key fields needs to have null values.  If it only iterates up to num_keys there won't be any value for non key fields.

Reply:
That's taken care of by the resize call above - it initializes the entire array to false

Reply:
But it's not affecting the builder. The new record is created from builder. One has to populate all fields in builder.